pyriemann/embedding.py

```diff
@@ -282,6 +298,10 @@ def barycenter_weights(X, Y, indices, metric="riemann", reg=1e-3):
         Indices of the points in Y used to compute the barycenter
     metric : {"euclid", "logeuclid", "riemann"}, default="riemann"
         Kernel metric.
+    kernel : callable, default=None
+        Kernel function to use for the embedding. If None, the canonical
+        kernel specified by the metric is used. Must be a function that
+        takes the arguments (X, Cref, metric).
```

Comment:
Since function `pyriemann.utils.kernel.kernel` allows a callable `metric`,
we can simply update the parameter `metric` as:
```suggestion
    metric : string | callable, default="riemann"
        Metric of kernel used for embedding, can be:
        "euclid", "logeuclid", "riemann", or a callable function.
```
Same remark for all new `kernel` parameters.

Would that cover your use case?

Reply:
@qbarthelemy with the new extension of the kernel module I have planned, this would not work. It will also cover distance kernel etc so specifying a metric should then only affect the distance metric.

Reply:
I understand. 

But it's difficult to judge which is the best solution since I don't see the code of the future kernel module.

Reply:
@qbarthelemy Here is the preliminary code https://github.com/gabelstein/pyRiemann/tree/new_kernels

Reply:
In the new module it would make most sense to make ktype be able to be the callable kernel function.

Reply:
In this case, I prefer to rename `kernel` in `ktype`.

Reply:
Wouldn't that be confusing if the ktype in the kernel module means something else than the ktype in the embedding? 
In the kernel module, ktype is either a string or a callable that does something with the metric (string).
Here, we need a kernel method to be passed to LLE.

Reply:
You are right, let's keep `kernel`.